### ISSUE TITLE
Can't see the GitLab module, but the plug-in list shows correct 

### ISSUE THREAD

**Issue by vnovakovic** [btey/openproject-gitlab-integration#5]
Can't see the GitLab module in OpenProject, but the plug-in lists shows that the module has been installed and available.

Can't choose GitLab module:
![Screenshot 2022-01-24 at 15 14 45](https://user-images.githubusercontent.com/10740860/150799847-5e1199c6-1901-4e7b-8d99-5149003625d6.png)

but the Plugin list shows it correctly:
![Screenshot 2022-01-24 at 15 21 05](https://user-images.githubusercontent.com/10740860/150800179-fcbaa0f9-e2ab-490c-85d2-cd06f773cd76.png)

Any idea or suggestion what to do?

KR,
V.

**Comment by btey** [comment]
The Gitlab plugin follows the same permissions model as Github. By default, Github is activated but in this case you have to activate the permissions to see it. You can go to Administration and in the Roles management, select the desired Role and activate the Gitlab checkbox.

Regards,
Ben

**Comment by btey** [comment]
I don't know what version of OpenProject you have installed. I plublished a new pre-release version v2.0.1 just for OpenProject 12 (v12.0.7), not for OpenProject 11. You can try this one. In short, in a manual installation scenario, the steps to add the plugin are:
- modify Gemfile.modules and Gemfile.lock as described in the readme file
- precompiled assets (for example, RAILS_ENV="production" ./bin/rake assets:precompile)
- restart apache

Regards,
Ben

**Comment by btey** [comment]
Hi @vnovakovic!

I have published a new version v2.0.2 with the updated Readme that indicates two changes that must be made in the OpenProject code for the plugin to work. Try this when you have a moment, it might be the cause of your problem.

Thank you!
Ben

**Comment by vnovakovic** [comment]
Hi Ben,

I have tried also with v2.0.2 and after restarting the OpenProject, puma service is not strating. If try to do on my own like:

openproject run puma

I'm getting the following message:

You are trying to install in deployment mode after changing
your Gemfile. Run `bundle install` elsewhere and add the
updated Gemfile.lock to version control.

If this is a development machine, remove the /opt/openproject/Gemfile freeze
by running `bundle config unset deployment`.

The dependencies in your gemfile changed
The gemspecs for path gems changed

You have added to the Gemfile:
* openproject-gitlab_integration

root@project01:/opt/openproject# bundle config unset deployment
root@project01:/opt/openproject# openproject run puma
Traceback (most recent call last):
	12: from /opt/openproject/vendor/bundle/bin/puma:29:in `<main>'
	11: from /opt/openproject/vendor/bundle/bin/puma:29:in `load'
	10: from /opt/openproject/vendor/bundle/ruby/2.7.0/gems/puma-5.5.2/bin/puma:8:in `<top (required)>'
	 9: from /opt/openproject/vendor/bundle/ruby/2.7.0/gems/puma-5.5.2/bin/puma:8:in `new'
	 8: from /opt/openproject/vendor/bundle/ruby/2.7.0/gems/puma-5.5.2/lib/puma/cli.rb:71:in `initialize'
	 7: from /opt/openproject/vendor/bundle/ruby/2.7.0/gems/puma-5.5.2/lib/puma/cli.rb:71:in `new'
	 6: from /opt/openproject/vendor/bundle/ruby/2.7.0/gems/puma-5.5.2/lib/puma/launcher.rb:59:in `initialize'
	 5: from /opt/openproject/vendor/bundle/ruby/2.7.0/gems/puma-5.5.2/lib/puma/configuration.rb:216:in `load'
	 4: from /opt/openproject/vendor/bundle/ruby/2.7.0/gems/puma-5.5.2/lib/puma/configuration.rb:216:in `each'
	 3: from /opt/openproject/vendor/bundle/ruby/2.7.0/gems/puma-5.5.2/lib/puma/configuration.rb:216:in `block in load'
	 2: from /opt/openproject/vendor/bundle/ruby/2.7.0/gems/puma-5.5.2/lib/puma/dsl.rb:81:in `_load_from'
	 1: from /opt/openproject/vendor/bundle/ruby/2.7.0/gems/puma-5.5.2/lib/puma/dsl.rb:81:in `instance_eval'
config/puma.rb:6:in `_load_from': uninitialized constant #<Class:#<Puma::DSL:0x00005587b5699618>>::OpenProject (NameError)

I do not have experience with ruby. This is installation under Ubuntu 20.4.03 LTS.

Thank you for the update.

KR,
Vlad

**Comment by btey** [comment]
Hi Vladimir!

I have created a VM with Ubuntu 20.04 LTS. After installing the openproject package (only install not setup), you will have to install nodejs/npm. I have used Ubuntu's own:

`$ sudo apt install nodejs npm`

This is necessary to precompile the assets, otherwise only the webhooks work for you but you will not see the GITLAB tab.

Without actually running the openproject setup, first copy the plugin code into place, modify Gemfile (lock and modules) and the two OpenProject code files as indicated in the Readme. With those changes execute the configuration:

```
$ sudo openproject run bundle config unset deployment
$ sudo openproject configure
$ sudo openproject run bundle config set deployment 'true'
```

When the process finishes OpenProject should be running. But the pre-compilation of the assets will be missing. For that you first need to install the node modules that OpenProject needs. I got it with the command:

```
$ cd /opt/openproject
$ sudo npm install
```

The first time it gave me module integrity errors, but I ran it again and it finished fine. The next step is to precompile the assets. I did it with the command:

```
$ cd /opt/openproject
$ sudo openproject run bundle exec rake assets:precompile
```

When finished restart the OpenProject service, and it should work. After activating the plugin in the project and giving permissions to the user role you should see the GITLAB tab. These are some screenshots of the plugin working:

![Screen Shot 2022-02-13 at 8 13 57 PM](https://user-images.githubusercontent.com/14983519/153784478-43144289-e358-4690-ba3c-43c25fe56bc3.png)

![Screen Shot 2022-02-13 at 8 14 38 PM](https://user-images.githubusercontent.com/14983519/153784514-fd9ae2bd-c6d1-441e-91c8-28f0cafdb815.png)

![Screen Shot 2022-02-13 at 8 15 07 PM](https://user-images.githubusercontent.com/14983519/153784543-c5b8574d-57d8-4d4d-bda5-08078086642a.png)

![Screen Shot 2022-02-13 at 8 15 40 PM](https://user-images.githubusercontent.com/14983519/153784632-54db96f3-c907-4476-85a2-864f213a7a5a.png)

I hope it is useful (and not leave me any step).

Ben

**Comment by btey** [comment]
With the help of OpenProject Team I released a new version v2.0.3 that no longer requires changes in the source code of OpenProject Core.

**Comment by vnovakovic** [comment]
great news about v2.0.3, I will try it right now.

I spend last night using your clean and detail instruction for Ubuntu (appreciate efforts that you did, thank you very much), but I face some issues even with it :-(

So, today I prepared CentOS 8 VM to try the baseline patching there later in the evening.

But now in meantime, I will try one more time the implementation on running Ubuntu with v2.0.3  

Nice to hear that you have support from OpenProject team. GitLab support is a good step for OpenProject.

**Comment by vnovakovic** [comment]
Hi Ben,

I have succeeded to get Ubuntu and v2.0.3 working together using your instructions.

The only correction that I would suggest in your Ubuntu installtion guide would be to update the line:

`$ cd /opt/openproject
$ sudo npm install`
 
 to:
 
`$ cd /opt/openproject/frontend
$ sudo npm install`

I got this warning npm WARN lifecycle openproject@0.1.0~postinstall: cannot run in wd openproject@0.1.0 cd frontend && npm install (wd=/opt/openproject)

So, instead to use working directory: /opt/openproject it is needed to enter the **frontend** directory to run **npm install** command.

Basically, you can close this open issue as successful.

Few more action points I have on my side:

1. To migate Postgresql data from the old server and check if everything is working fine.
2. Configure webhooks between OpenProject and GitLab.
3. To check with upcoming OpenProject upgrade will GitLab module will stay intact and functional.

Thank you for your great module and provided instructions. I hope that OpenProject team will soon include your GitLab module as a part of standard OpenProject distribution. This would be smart and cool.
